### PR TITLE
Contract Upgrade Proposal Form Integration

### PR DESCRIPTION
**Pull request by roshan-bisht-lab49**
Signed-off-by: Roshan Singh Bisht <roshan.bisht@lab49.com>

### CODE REVIEW

File: src/dex-ui/services/HederaService/GovernorService.ts [hunk 1]
```diff
@@ -165,6 +169,44 @@ const sendCreateTransferTokenProposalTransaction = async (
...
+ * @returns
+ */
+const sendCreateContractUpgradeProposalTransaction = async (
+  params: CreateContratctUpgradeProposalParams
+): Promise<TransactionResponse> => {
+  const { title, description, linkToDiscussion, contarctId, proxyId, signer } = params;
+  const upgradeProposalContractId = ContractId.fromString(contarctId).toSolidityAddress();
+  const upgradeProposalProxyId = ContractId.fromString(proxyId).toSolidityAddress();
+
+  console.log(`Roshan data is ${title} ${upgradeProposalContractId} ${upgradeProposalProxyId} ${signer}`);
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/services/HederaService/GovernorService.ts:193]: Looks like there is a left over console.log here.

File: src/dex-ui/pages/CreateProposal/ProposalForms/ContractUpgradeProposalForm.tsx [hunk 2]
```diff
@@ -1,16 +1,200 @@
...
+        </Flex>
+      </form>
+      <LoadingDialog
+        isOpen={deployContract.isError}
+        message={deployContract.error?.message ?? ""}
+        icon={<WarningIcon color="#EF5C5C" h={10} w={10} />}
+        buttonConfig={{
+          text: "Dismiss",
+          onClick: () => {
+            resetFormValues();
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/pages/CreateProposal/ProposalForms/ContractUpgradeProposalForm.tsx:181]: Is clearing out the form on a failed transaction the desired UX? I would double check with Anna.

File: src/dex-ui/pages/CreateProposal/ProposalForms/ContractUpgradeProposalForm.tsx [hunk 3]
```diff
@@ -1,16 +1,200 @@
...
+              height="40px"
+              isLoading={isSubmitting}
+              alignSelf="end"
+            >
+              Publish
+            </Button>
+          </Flex>
+        </Flex>
+      </form>
+      <LoadingDialog
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/pages/CreateProposal/ProposalForms/ContractUpgradeProposalForm.tsx:174]: In the future, you can use a single `LoadingDialog` component for loading and a single `LoadingDialog` component for error states. Take a look at `ProposalDetails` component and `useProposalDetails` hook to see how we are currently doing this. We can enhance the LoadingDialog component in the future to directly accept getter functions for **isOpen** and **message** values.

**Review by damiano-melcarne-lab49 — CHANGES_REQUESTED** [review]

File: src/dex-ui/services/HederaService/GovernorService.ts (hunk 1 above)
Comment (roshan-bisht-lab49) [reply src/dex-ui/services/HederaService/GovernorService.ts]: Done

File: src/dex-ui/pages/CreateProposal/ProposalForms/ContractUpgradeProposalForm.tsx (hunk 2 above)
Comment (roshan-bisht-lab49) [reply src/dex-ui/pages/CreateProposal/ProposalForms/ContractUpgradeProposalForm.tsx]: I will check with here

File: src/dex-ui/pages/CreateProposal/ProposalForms/ContractUpgradeProposalForm.tsx (hunk 3 above)
Comment (roshan-bisht-lab49) [reply src/dex-ui/pages/CreateProposal/ProposalForms/ContractUpgradeProposalForm.tsx]: Done

**Review by damiano-melcarne-lab49 — APPROVED** [review]

**Comment by roshan-bisht-lab49** [comment]
Merging this PR to complete the rest `description` and `link to discussion` work.

**Pull request merged**